packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-11/zoe-full-upgrade/run-prober-script.js

```diff
@@ -1,7 +1,8 @@
-// to turn on ts-check:
+// @ts-no-check
 /* global E */
 
-// import { E } from "@endo/far";
+// to turn on ts-check:
```

Comment:
I don't get it. Why disable ts-check if this is all it takes to enable it?  Oh, does the `import` fail at runtime?  if that's it then please enable ts-check and do a type-only import.
```js
/** @typedef {import('@endo/far').E} E */
```

Reply:
@dckc I'm just cargo-culting here. What's your advice on best practices?

Reply:
> type-only import

TIL. Nifty.

But at runtime, the string `import(` cannot occur anywhere in the script, even in comments. See [rejectImportExpressions](https://github.com/endojs/endo/blob/ebc8f66e9498f13085a8e64e17fc2f5f7b528faa/packages/ses/src/transforms.js#L143)

>  Conservatively reject the source text if it may contain a dynamic
>  import expression. To reject without parsing, `rejectImportExpressions` will
>  also reject some other text as well.

Reply:
oh... this `import(` limitation is another thing that's addressed by the "cadillac method", using `@agoric/deploy-script-support`:

https://github.com/Agoric/agoric-sdk/blob/8bcf6af10ce538871cc0bf5798000247c367313a/packages/deploy-script-support/src/code-gen.js#L20

Reply:
by way of documentation, I started...

 - [Why does import cause errors in a permissioned upgrade \(CoreEval\) script? Is there a better way? · Agoric/agoric\-sdk · Discussion \#8087](https://github.com/Agoric/agoric-sdk/discussions/8087)